OSSM-8200 Document how to scope the mesh with Discovery Selectors (#84469)
openshift/openshift-docs: modules/ossm-about-discoveryselectors.adoc

// Module included in the following assemblies:
// install/ossm-installing-openshift-service-mesh.adoc

:_mod-docs-content-type: Concept
[id="ossm-about-discoveryselectors_{context}"]
= About DiscoverySelectors
:context: ossm-about-discoveryselectors

`discoverySelectors` let the mesh administrator control which namespaces the control plane can access. Using a {k8s} label selector, the administrator sets the criteria for the namespaces visible to the control plane, excluding any namespaces that do not match the specified criteria.

> 

Reviewer: Avoid placing multiple notes next to each other. If multiple notes are necessary, restructure the information by moving the less-important statements into the flow of the main content. See https://redhat-documentation.github.io/supplementary-style-guide/#admonitions

Author: Moved the less important note below.

[NOTE]
====

> Istiod will always open a watch to {ocp-short-name} for all namespaces. However, `discoverySelectors` will ignore objects that are not selected very early in its processing, minimizing costs.

Reviewer: ```suggestion
Istiod always opens a watch to {ocp-short-name} for all namespaces. However, discovery selectors ignore objects that are not selected early in its processing, minimizing costs.
```

====

[NOTE]
====

> `discoverySelectors` is not a security boundary. Istiod will continue to have access to all namespaces even when you have configured your `discoverySelector`.

Reviewer 2: This note is not correct.

> `discoverySelectors` is not a security boundary. Istiod will continue to have access to all namespaces even when you have configured your `discoverySelector`.

Reviewer: ```suggestion
The `discoverySelectors` field is not a security boundary. Istiod continues to have access to all namespaces even when you have configured the `discoverySelector` field.
```
or 
```suggestion
Discovery selectors are not a security boundary. Istiod continues to have access to all namespaces even when you have configured the `discoverySelector` field.
```

Author: Took the second suggestion.

====

> The `discoverySelectors` field accepts an array of {k8s} selectors, which apply to labels on namespaces. You can configure each selector for different use cases, such as:

Reviewer: ```suggestion
The `discoverySelectors` field accepts an array of {k8s} selectors, which apply to labels on namespaces. You can configure each selector for different use cases:
```

* Custom label names and values. For example, configure all namespaces with the label `istio-discovery=enabled`.

> * A list of namespace labels using set-based selectors with OR logic. For instance, configure namespaces with `istio-discovery=enabled` OR `region=us-east1`.

Reviewer: ```suggestion
* A list of namespace labels by using set-based selectors with OR logic. For instance, configure namespaces with `istio-discovery=enabled` OR `region=us-east1`.
```
or 
```suggestion
* A list of namespace labels that uses set-based selectors with OR logic. For instance, configure namespaces with `istio-discovery=enabled` OR `region=us-east1`.
```

Author: Took the first suggestion.
